src/main/resources/assets/admin/common/js/form/InputView.ts

```diff
@@ -117,15 +118,45 @@ export class InputView
                 this.bottomButtonRow = new DivEl('bottom-button-row');
                 this.appendChild(this.bottomButtonRow);
                 this.bottomButtonRow.appendChild(this.addButton);
+
+                this.toggleClass('one-and-only',
+                    this.input.getOccurrences().getMinimum() === 1 && this.input.getOccurrences().getMaximum() === 1);
+
+                inputTypeViewNotManagingAdd.onOccurrenceValueChanged(() => {
+                    this.toggleClass('has-invalid-user-input', !inputTypeViewNotManagingAdd.hasValidUserInput());
```

Comment:
This should be made into a separate method since it's used in 3 different handlers.